### DISCUSSION TITLE
Automap : can I exclude two tables from getting reflected

### DISCUSSION THREAD

**Discussion by emburse-manikamidha** [sqlalchemy/sqlalchemy#11506] [Usage Questions]
Hello

I have implemented Automap to auto detect existing tables in MySQL database and use them as models/ classes. Out of 700, 698 tables were correctly mapped. There are 2 tables that I cannot use even in my MySQL database. I get the same result in both Database and Automap (Base.prepare). 
Below is the warning : 
```
SAWarning: Skipping table dummy_table_name: Table or view named `dummy_table_name` could not be reflected: (pymysql.err.OperationalError) (1356, "View 'database.dummy_table_name' references invalid table(s) or column(s) or function(s) or definer/invoker of view lack rights to use them")
[SQL: DESCRIBE `dummy_table_name`]
(Background on this error at: https://sqlalche.me/e/20/e3q8)
  Base.prepare(autoload_with=engine, reflection_options={'views': True})
```
This issue exists and is valid. I just don't want this warning in my console. Is there a way to get Base.prepare to exclude the table dummy_table_name from mapping ?

Thanks

**Comment by CaselIT** [comment]
Hi,

You can provide `reflection_options` to prepare. These are option for MetaData.reflect, were you can pass `only` as a callable. See the documentation as additional details

**Reply by emburse-manikamidha** [reply]
@CaselIT I am using` Base.prepare(autoload_with=engine, reflection_options={'views': True}) ` to reflect the tables. I cannot use MetaData.reflect as I am using ORM approach.

**Reply by CaselIT** [reply]
You can pass a custom only callable in the reflection_option in `prepare()`

**Reply by emburse-manikamidha** [reply]
@CaselIT With 'only', it sounds like I would be able to pass the tables I want to reflect. Say, I have 700 tables in my database. I want to reflect all but two. I would like to just exclude two tables and not include 698 tables. Is there a way to exclude tables ?

**Reply by CaselIT** [reply]
As mentioned please refer to the documentation https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.MetaData.reflect.params.only

> If a callable is provided, it will be used as a boolean predicate to filter the list of potential table names. The callable is called with a table name and this MetaData instance as positional arguments and should return a true value for any table to reflect.
